tests/python/relay/test_json_runtime.py

```diff
@@ -0,0 +1,625 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Unit tests for JSON codegen and runtime."""
```

Comment:
This seems more like it's testing the DNNL codegen and runtime rather than the underlying JSON implementation, to the point no tests will run without DNNL. I think we should aim to steer away from this sort of approach and see if there are ways we can more directly test the implementation.

Reply:
The case you mentioned is indeed more desired, but that means we have to enhance the example c compiler to support all these ops, which seems impractical. That's why enabling DNNL in the CI is important.

Reply:
Yeah, I think this is actually a broader issue in TVM in that we don't seem to have a robust C++ unit testing framework so a lot of things are tested indirectly via Python. I'll have a think about this, but not blocking on this PR.